src/main/java/com/google/devtools/build/lib/remote/RemoteServerCapabilities.java

```diff
@@ -201,17 +201,24 @@ public static ClientServerCompatibilityStatus checkClientServerCompatibility(
         return result.build(); // No point checking other execution fields.
       }
 
-      // Check execution digest function.
-      if (execCap.getDigestFunction() == DigestFunction.Value.UNKNOWN) {
-        // Server side error -- this is not supposed to happen.
-        result.addError("Remote server error: UNKNOWN execution digest function.");
-      }
-      if (execCap.getDigestFunction() != digestFunction) {
+      // Check execution digest function. The protocol only later added
+      // support for multiple digest functions for remote execution, so
+      // check both the singular and repeated field.
+      if (execCap.getDigestFunctionsList().isEmpty() &&
```

Comment:
It misses a case where `execCap.getDigestFunctionsList().isEmpty()` and `execCap.getDigestFunction() == DigestFunction.Value.UNKNOWN`, it should print the UNKNOWN function, instead of empty list.

Reply:
The question is, is this really what should be preferred? My interpretation is that `digest_functions` is what's authoritative now, with `digest_function` being a fallback. If both of them are unset, then I would assume it would make most sense to display an error message using the native format of `digest_functions`, which is a list.

Reply:
Well, that make sense.